src/index.js

```diff
@@ -35,10 +32,10 @@ class Vault {
     return hmac.digest()
   }
 
-  async _derivedKey (salt) {
+  _derivedKey (salt) {
     if (!this[PASSWORD]) throw new Error('No password')
 
-    const derivedKey = await pbkdf2(this[PASSWORD], salt, 10000, 80, DIGEST)
+    const derivedKey = crypto.pbkdf2Sync(this[PASSWORD], salt, 10000, 80, DIGEST)
```

Comment:
This blocking op is penalizing everyone expecting to use a non-blocking async operation.

Reply:
Thanks. Done. I restructured the code instead. There isn't any sync/async bleed through any more. I restructured your code a little in order to prevent duplication.

Reply:
Thanks @commenthol . I've restructured your code for reuse (instead of copy & paste), there isn't any sync/async bleed through anymore.